### PR TITLE
[4656] Update HPITT import for general bulk import of trainees

### PR DESCRIPTION
**Pull request by AbigailMcP**
### Context

https://trello.com/c/ky539ZsV/4656-l-check-rake-tasks-work-for-bulk-uploading-trainees-bulk-upload

### Changes proposed in this pull request

* Update rake task to take the provider code as the first argument, so we can re-use this for general bulk import for any provider
* Fill in database fields for trainees that we infer for HPITT trainees:
  * Training route
  * Training initiatives
  * Funding type
  * Course UUID (if course code in csv is present)
* Support multiple comma-separate nationalities in csv
* Typo in `Course ITT Subject 3` heading name
* Copy across funding validation logic from original bulk import (code here: https://github.com/DFE-Digital/register-trainee-teachers/blob/main/app/lib/bulk_import.rb). This means that providers cannot give incorrect funding.
* Copy across progress progress validation logic from bulk import instead of just crudely setting all progress values to true
* Update the column headings so that HPITT csv and the bulk import csv match.
  * `Middle name` -> `Middle names`
  * `Last name` -> `Last names`
  * `Course ITT end date` -> `Course Expected End Date`
  - **Note: we will have to do this manually as the HPITT csv has already been sent to providers. But I didn't think it was worth building separate parsers for the sake of three heading differences.**

Not included: renaming the files to something more generic rather than scoped by `Hpitt`. I want to change the file names after this has been reviewed in case we lose the code diff.

### Guidance to review

To test out the import:
`rake "hpitt:import[provider_code,test.csv]"`

### Important business

- [ ] ~Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?~
- [ ] ~Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?~

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Comment by AbigailMcP** [comment]
[[L] Check rake tasks work for bulk uploading trainees (bulk upload)](https://trello.com/c/ky539ZsV/4656-l-check-rake-tasks-work-for-bulk-uploading-trainees-bulk-upload)

File: lib/tasks/hpitt_import.rake [hunk 1]
```diff
@@ -2,16 +2,18 @@
...
-  task :import, %i[csv_path] => [:environment] do |_, args|
+  task :import, %i[provider_code csv_path] => [:environment] do |_, args|
     csv = CSV.read(
       args.csv_path,
       headers: true,
       encoding: "ISO-8859-1",
       header_converters: ->(f) { f&.strip },
     )
 
+    provider = Provider.find_by!(code: args.provider_code)
```
Comment (markhallen) [inline lib/tasks/hpitt_import.rake:13]: Should we catch a potential typo in the `provider_code`?

**Comment by AbigailMcP** [comment]
@markhallen @edwardhorsford 
New bits in these commits:

f852025529f7c118c41da188c95fcc3c85a279a8
cc8297dba786f1037b1aae72367e3485e3a2df1c

**Review by markhallen — APPROVED** [review]
Oprah Crazy

**Pull request merged**